### PR TITLE
Replace WMI query with interop

### PR DESCRIPTION
**Pull request by aleksandrlevochkin**
**WI**
[AB#2142810](https://dev.azure.com/mseng/b924d696-3eae-4116-8443-9a18392d8544/_workitems/edit/2142810)

**Description**
When we turn on the `AZP_AGENT_CLEANUP_PSMODULES_IN_POWERSHELL` knob the WMI query times out for some machines.

I added a new knob that allows to use interop to find parent processes.

**Testing**

Tested scenarios (in PS Core on Windows, for other platforms there should be no need to cleanup psmodules):

1. `AZP_AGENT_CLEANUP_PSMODULES_IN_POWERSHELL: "true"`
  `AZP_AGENT_USE_INTEROP_TO_FIND_PARENT_PROCESS: "false"`
  WMI query is used to determine if powershell is one of the parent processes.

1. `AZP_AGENT_CLEANUP_PSMODULES_IN_POWERSHELL: "true"`
  `AZP_AGENT_USE_INTEROP_TO_FIND_PARENT_PROCESS: "true"`
  Native Windows function is used to determine if powershell is one of the parent processes.

### CODE REVIEW

**Review by DenisRumyantsev — APPROVED** [review]
I tested it. Changes work as expected. Looks good, thanks!

**Review by kirill-ivlev — APPROVED** [review]

File: src/Agent.Sdk/Util/ProcessUtil.cs [hunk 1]
```diff
@@ -1,16 +1,82 @@
...
+
+        internal static (Process process, Dictionary<string, string> telemetry) GetParentProcess(Process process)
+        {
+            var telemetry = new Dictionary<string, string>();
+
+            try
+            {
+                int? parentProcessId = GetParentProcessId(process.Handle);
+
+                if (parentProcessId == null) return (null, telemetry);
```
Comment (DergachevE) [inline src/Agent.Sdk/Util/ProcessUtil.cs:55]: nitpick: parentheses

Comment (aleksandrlevochkin) [reply src/Agent.Sdk/Util/ProcessUtil.cs]: Fixed

File: src/Agent.Sdk/Util/ProcessUtil.cs [hunk 2]
```diff
@@ -1,16 +1,82 @@
...
     {
+        private const string ParentProcessState = nameof(ParentProcessState);
+
+        internal static int? GetParentProcessId(IntPtr handle)
+        {
+            Interop.PROCESS_BASIC_INFORMATION pbi;
+            int returnLength;
+            int status = Interop.NtQueryInformationProcess(handle, 0, out pbi, Marshal.SizeOf<Interop.PROCESS_BASIC_INFORMATION>(), out returnLength);
+
+            if (status != 0) return null;
```
[the remark this answers is not in the document]
Comment (aleksandrlevochkin) [reply src/Agent.Sdk/Util/ProcessUtil.cs]: Fixed

**Review by DergachevE — APPROVED** [review]

**Pull request merged**